go/api/commands.go

```diff
@@ -647,3 +647,281 @@ type HashCommands interface {
 	// [valkey.io]: https://valkey.io/commands/hstrlen/
 	HStrLen(key string, field string) (Result[int64], error)
 }
+
+// SetCommands supports commands and transactions for the "Set Commands" group for standalone and cluster clients.
+//
+// See [valkey.io] for details.
+//
+// [valkey.io]: https://valkey.io/commands/?group=set
+type SetCommands interface {
+	// SAdd adds specified members to the set stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key     - The key where members will be added to its set.
+	//  members - A list of members to add to the set stored at key.
+	//
+	// Return value:
+	//  The Result[int64] containing number of members that were added to the set,
+	//  or [api.NilResult[int64]](api.CreateNilInt64Result()) when the key does not exist.
+	//
+	// For example:
+	//  result, err := client.SAdd("my_set", []string{"member1", "member2"})
+	//  // result.Value(): 2
+	//  // result.IsNil(): false
+	//
+	// [valkey.io]: https://valkey.io/commands/sadd/
+	SAdd(key string, members []string) (Result[int64], error)
+
+	// SRem removes specified members from the set stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key     - The key from which members will be removed.
+	//  members - A list of members to remove from the set stored at key.
+	//
+	// Return value:
+	//  The Result[int64] containing number of members that were removed from the set, excluding non-existing members.
+	//  Returns [api.NilResult[int64]](api.CreateNilInt64Result()) if key does not exist.
+	//
+	// For example:
+	//  result, err := client.SRem("my_set", []string{"member1", "member2"})
+	//  // result.Value(): 2
+	//  // result.IsNil(): false
+	//
+	// [valkey.io]: https://valkey.io/commands/srem/
+	SRem(key string, members []string) (Result[int64], error)
+
+	// SMembers retrieves all the members of the set value stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//   key - The key from which to retrieve the set members.
+	//
+	// Return value:
+	//   A map[Result[string]]struct{} containing all members of the set.
+	//   Returns an empty map if key does not exist.
+	//
+	// For example:
+	//   // Assume set "my_set" contains: "member1", "member2"
+	//   result, err := client.SMembers("my_set")
+	//   // result equals:
+	//   // map[Result[string]]struct{}{
+	//   //   api.CreateStringResult("member1"): {},
+	//   //   api.CreateStringResult("member2"): {}
+	//   // }
+	//
+	// [valkey.io]: https://valkey.io/commands/smembers/
+	SMembers(key string) (map[Result[string]]struct{}, error)
```

Comment:
I understand that we are using `map[Result[string]]struct{}` to represent `Set` here but I think we can simply return a `slice of Result[string]` from here.